### PR TITLE
Broker/Server separate configs

### PR DESCRIPTION
**Pull request by fboucquez**
Splitting `userconfig` into `server-config` and `broker-config` for each service.
Improved server/broker start scripts.
Improved recovery? @Wayonb 

https://github.com/nemtech/symbol-bootstrap/issues/108

### CODE REVIEW

File: config/docker/server/startBroker.sh.mustache [hunk 1]
```diff
@@ -1,19 +1,92 @@
 #!/bin/bash
```
Comment (Wayonb) [inline config/docker/server/startBroker.sh.mustache:1]: Why combined all the scripts?

File: config/docker/server/startBroker.sh.mustache [hunk 2]
```diff
@@ -1,19 +1,92 @@
...
+rm -f "{{{dataDirectory}}}/$application-recovery.started"
+
 if [ "$mode" == "DEBUG" ]; then
   echo "Setting up core dump..."
   mkdir -p ./logs
-  echo './logs/broker.%e.%t' > /proc/sys/kernel/core_pattern
+  echo "./logs/$application.%e.%t" >/proc/sys/kernel/core_pattern
+fi
+
+if [ "$waitForOther" == "true" ]; then
```
Comment (Wayonb) [inline config/docker/server/startBroker.sh.mustache:43]: Is the wait needed for the broker?

File: config/docker/server/startServer.sh.mustache [hunk 3]
```diff
@@ -1,17 +1,92 @@
...
+
 if [ "$mode" == "DEBUG" ]; then
   echo "Setting up core dump..."
   mkdir -p ./logs
-  echo './logs/server.%e.%t' >/proc/sys/kernel/core_pattern
+  echo "./logs/$application.%e.%t" >/proc/sys/kernel/core_pattern
 fi
 
-exec env LD_LIBRARY_PATH={{{catapultAppFolder}}}/lib:{{{catapultAppFolder}}}/deps {{{catapultAppFolder}}}/bin/catapult.server ./userconfig
+if [ "$waitForOther" == "true" ]; then
```
Comment (Wayonb) [inline config/docker/server/startServer.sh.mustache:43]: The wait should be after the recovery is run. If the other application has started the recovery will fail.

**Review by Wayonb — CHANGES_REQUESTED** [review]

File: config/docker/server/startBroker.sh.mustache (hunk 1 above)
Comment (fboucquez) [reply config/docker/server/startBroker.sh.mustache:1]: I've merged together the recovery script and the start script for each service. I think it gives better and conrtol visibility. The service's run will stay the. Broker and Server scripts are pretty much the same, I'm thinking about merging them together into one.

File: config/docker/server/startBroker.sh.mustache (hunk 2 above)
Comment (fboucquez) [reply config/docker/server/startBroker.sh.mustache:43]: It's not needed. I've kept it so we can potentially have a common start.sh file.

File: config/docker/server/startServer.sh.mustache (hunk 3 above)
Comment (fboucquez) [reply config/docker/server/startServer.sh.mustache:43]: Thanks, I'll move. it.
Question. Do you have a strategy about how to stop one service when the other one fails? If both catapult.broker and catapult.,server are running, and one stops. How to kill the other and clean up the flags so both services start (and recover) from scratch.

I'm thinking about pooling for the "theOtherApplication.started". It this file is removed (the other processed stoped), kill the current process. Do you have something better? Another alternative is using compose's health checks. But I don't want to ping the server.

Comment (Wayonb) [reply config/docker/server/startServer.sh.mustache]: Push some changes that will do the automatic recovery the api node.  Take a look....It works but needs some cleanup.

Comment (fboucquez) [reply config/docker/server/startServer.sh.mustache]: Thanks @Wayonb. I looks really nice.

I've merged the 2 scripts together into start.sh. It's not a mustache template anymore. I've added -f to the `rm` commands. The start.sh could be used potentially for non-docker-compose node admins. 

Could you validate https://github.com/nemtech/symbol-bootstrap/blob/broker_server_config/config/docker/server/start.sh

File: config/docker/server/start.sh [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+touch "$dataDirectory/$application.started"
+
+processName="catapult.$application"
+echo "!!!! Starting $application...."
+$catapultAppFolder/bin/$processName "$config" &
+export applicationPid=$!
+
+cleanup() {
+  echo "Shutting down $application"
+  echo "kill -s SIGTERM $applicationPid"
```
Comment (Wayonb) [inline config/docker/server/start.sh:97]: missed removing the debug echo. -  `echo "kill -s SIGTERM $applicationPid"`

File: config/docker/server/start.sh [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    echo "The process $processName has already exited."
+    wait $applicationPid
+    exitStatus=$?
+    rm -f "./data/$application.started"
+    exit $exitStatus
+  fi
+
+  if [ -e $otherApplicationRecoveryFile ]; then
+    echo "Other process waiting for recovery. Shutting down $application"
+    echo "kill -s SIGTERM $applicationPid"
```
Comment (Wayonb) [inline config/docker/server/start.sh:121]: Here also.

**Review by Wayonb — DISMISSED** [review]

**Pull request merged**